api4/webhook_test.go

```diff
@@ -169,6 +168,20 @@ func TestGetIncomingWebhooks(t *testing.T) {
 		t.Fatal("no hooks should be returned")
 	}
 
+	// Test Team User Case: Permission to MANAGE_INCOMING_WEBHOOKS but not to MANAGE_OTHERS_INCOMING_WEBHOOKS
+	th.AddPermissionToRole(model.PERMISSION_MANAGE_INCOMING_WEBHOOKS.Id, model.TEAM_USER_ROLE_ID)
+
+	hooks, resp = Client.GetIncomingWebhooksForTeam(th.BasicTeam.Id, 0, 1000, "")
+	CheckNoError(t, resp)
+	assert.Equal(t, len(hooks), 0)
```

Comment:
Could we expand the test data so that we verify that we get our own hooks, and then others hooks when we have the extra permission below?

Reply:
✅I've extracted the unit test for the FilterByTeamByUser to its own case and added the extra testing data suggested.